pkg/monitor/monitor_test.go

```diff
@@ -277,3 +283,62 @@ func testOfflineStorageNodeDuplicateIP(t *testing.T) {
 	_, err = k8s.Instance().GetPodByName(pod.Name, "")
 	require.NoError(t, err, "expected no error from get pod as pod should not be deleted")
 }
+
+func testVolumeAttachmentCleanup(t *testing.T) {
+	nodeToKeepOnline := nodes.Items[0].Name
+	nodeToTakeOffline := nodes.Items[1].Name
+
+	// Create multiple pods on different nodes, some with volumeattachments, some without.
+	// Stop the driver on the node with the attachment and make sure only that pod and volumeattachment are deleted.
+
+	// Create two pods on node N0 that will remain healthy. One attached, one not.
+	healthyPodAttached := newPod("testVolumeAttachmentCleanupHealtyAttached", []string{driverVolumeName})
+	healthyPodAttached.Spec.NodeName = nodeToKeepOnline
+	_, err := k8s.Instance().CreatePod(healthyPodAttached)
+	require.NoError(t, err, "failed to create pod")
+
+	healthyPodDetached := newPod("testVolumeAttachmentCleanHealtyupDetached", []string{})
+	healthyPodDetached.Spec.NodeName = nodeToKeepOnline
+	_, err = k8s.Instance().CreatePod(healthyPodDetached)
+	require.NoError(t, err, "failed to create pod")
+
+	// Create two pods on node N1 that will be taken offline temporarily. One attached, one not.
+	unhealthyPodAttached := newPod("testVolumeAttachmentCleanupUnheathyAttached", []string{attachmentVolumeName})
+	unhealthyPodAttached.Spec.NodeName = nodeToTakeOffline
+	_, err = k8s.Instance().CreatePod(unhealthyPodAttached)
+	require.NoError(t, err, "failed to create pod")
+
+	unhealthyPodDetached := newPod("testVolumeAttachmentCleanupUnheathyDetached", []string{})
+	unhealthyPodDetached.Spec.NodeName = nodeToTakeOffline
+	_, err = k8s.Instance().CreatePod(unhealthyPodDetached)
+	require.NoError(t, err, "failed to create pod")
+
+	// Kill N1
+	err = driver.UpdateNodeStatus(1, volume.NodeOffline)
+	require.NoError(t, err, "Error setting node status to Offline")
+	defer func() {
+		err = driver.UpdateNodeStatus(0, volume.NodeOnline)
+		require.NoError(t, err, "Error setting node status to Online")
+	}()
+
+	// VolumeAttachments (VA) for N1 should be deleted, but VA for N0 should remain.
+	// HOWEVER, we can't currently test this because I didn't add CreateVolumeAttachment to sched-ops
```

Comment:
Can you add that? The test isn't really testing the fix without that

Reply:
Yeah, in the process of adding it now